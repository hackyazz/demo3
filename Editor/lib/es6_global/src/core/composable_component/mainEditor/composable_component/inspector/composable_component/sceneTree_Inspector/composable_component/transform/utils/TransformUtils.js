

import * as Curry from "../../../../../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as AppStore$WonderEditor from "../../../../../../../../../ui/store/AppStore.js";
import * as FloatService$WonderEditor from "../../../../../../../../../../service/atom/FloatService.js";
import * as UIStateService$WonderEditor from "../../../../../../../../../../service/state/ui/UIStateService.js";
import * as StateEditorService$WonderEditor from "../../../../../../../../../../service/state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/state/StateEngineService.js";
import * as SceneTreeEditorService$WonderEditor from "../../../../../../../../../../service/state/editor/sceneTree/SceneTreeEditorService.js";
import * as TransformEditorService$WonderEditor from "../../../../../../../../../../service/state/editor/transform/TransformEditorService.js";
import * as TransformEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/TransformEngineService.js";
import * as GameObjectComponentEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/gameObject/GameObjectComponentEngineService.js";

function truncateTransformValue(param) {
  return /* tuple */[
          FloatService$WonderEditor.truncateFloatValue(param[0], 5),
          FloatService$WonderEditor.truncateFloatValue(param[1], 5),
          FloatService$WonderEditor.truncateFloatValue(param[2], 5)
        ];
}

var getSceneTreeNodeLocalPosition = TransformEngineService$WonderEditor.getLocalPosition;

function getTransformPositionData(transformComponent, engineState) {
  return truncateTransformValue(TransformEngineService$WonderEditor.getLocalPosition(transformComponent, engineState));
}

function getTransformScaleData(transformComponent, engineState) {
  return truncateTransformValue(TransformEngineService$WonderEditor.getLocalScale(transformComponent, engineState));
}

function getTransformRotationData(transformComponent, engineState) {
  var match = TransformEditorService$WonderEditor.getLocalEulerAngleAndInit(transformComponent, /* tuple */[
        StateEditorService$WonderEditor.getState(/* () */0),
        engineState
      ]);
  StateEditorService$WonderEditor.setState(match[1]);
  return truncateTransformValue(match[0]);
}

function refreshTransformWithDispatchFunc(dispatchFunc, param) {
  var engineState = param[1];
  var editorState = param[0];
  var editorState$1 = TransformEditorService$WonderEditor.removeLocalEulerAngleData(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(SceneTreeEditorService$WonderEditor.unsafeGetCurrentSceneTreeNode(editorState), engineState), editorState);
  StateEditorService$WonderEditor.setState(editorState$1);
  StateEngineService$WonderEditor.setState(engineState);
  Curry._1(dispatchFunc, [
        AppStore$WonderEditor.UpdateAction,
        /* Update */[/* array */[/* Inspector */2]]
      ]);
  return /* tuple */[
          StateEditorService$WonderEditor.getState(/* () */0),
          StateEngineService$WonderEditor.unsafeGetState(/* () */0)
        ];
}

function refreshTransform(param) {
  return refreshTransformWithDispatchFunc(UIStateService$WonderEditor.getDispatch(/* () */0), /* tuple */[
              param[0],
              param[1]
            ]);
}

export {
  truncateTransformValue ,
  getSceneTreeNodeLocalPosition ,
  getTransformPositionData ,
  getTransformScaleData ,
  getTransformRotationData ,
  refreshTransformWithDispatchFunc ,
  refreshTransform ,
  
}
/* AppStore-WonderEditor Not a pure module */
