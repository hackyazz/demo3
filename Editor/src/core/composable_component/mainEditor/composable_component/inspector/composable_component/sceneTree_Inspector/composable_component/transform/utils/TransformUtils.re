let truncateTransformValue = ((x, y, z)) => {
  let truncateLen = 5;
  (
    FloatService.truncateFloatValue(x, truncateLen),
    FloatService.truncateFloatValue(y, truncateLen),
    FloatService.truncateFloatValue(z, truncateLen),
  );
};

let getSceneTreeNodeLocalPosition = (transformComponent, engineState) =>
  TransformEngineService.getLocalPosition(transformComponent, engineState);

let getTransformPositionData = (transformComponent, engineState) =>
  getSceneTreeNodeLocalPosition(transformComponent, engineState)
  |> truncateTransformValue;

let getTransformScaleData = (transformComponent, engineState) =>
  TransformEngineService.getLocalScale(transformComponent, engineState)
  |> truncateTransformValue;

let getTransformRotationData = (transformComponent, engineState) => {
  let (data, editorState) =
    TransformEditorService.getLocalEulerAngleAndInit(
      transformComponent,
      (StateEditorService.getState(), engineState),
    );

  editorState |> StateEditorService.setState |> ignore;

  data |> truncateTransformValue;
};

let refreshTransformWithDispatchFunc =
    (dispatchFunc, (editorState, engineState)) => {
  let editorState =
    TransformEditorService.removeLocalEulerAngleData(
      GameObjectComponentEngineService.unsafeGetTransformComponent(
        SceneTreeEditorService.unsafeGetCurrentSceneTreeNode(editorState),
        engineState,
      ),
      editorState,
    );

  editorState |> StateEditorService.setState |> ignore;
  engineState |> StateEngineService.setState |> ignore;

  dispatchFunc(AppStore.UpdateAction(Update([|UpdateStore.Inspector|])))
  |> ignore;

  (StateEditorService.getState(), StateEngineService.unsafeGetState());
};

let refreshTransform = ((editorState, engineState)) =>
  refreshTransformWithDispatchFunc(
    UIStateService.getDispatch(),
    (editorState, engineState),
  );