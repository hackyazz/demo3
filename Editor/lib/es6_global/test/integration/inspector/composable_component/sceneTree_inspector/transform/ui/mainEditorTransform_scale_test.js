

import * as Curry from "../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as ReactTool$WonderEditor from "../../../../../../tool/ui/ReactTool.js";
import * as FloatInputTool$WonderEditor from "../../../../../tool/FloatInputTool.js";
import * as GameObjectTool$WonderEditor from "../../../../../../tool/GameObjectTool.js";
import * as TransformUtils$WonderEditor from "../../../../../../../src/core/composable_component/mainEditor/composable_component/inspector/composable_component/sceneTree_Inspector/composable_component/transform/utils/TransformUtils.js";
import * as DirectorToolEngine$WonderEditor from "../../../../../../tool/engine/DirectorToolEngine.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../../tool/MainEditorSceneTool.js";
import * as MainEditorTransformTool$WonderEditor from "../tool/MainEditorTransformTool.js";
import * as MainEditorTransformTestTool$WonderEditor from "../tool/MainEditorTransformTestTool.js";
import * as NoWorkerJobConfigToolEngine$WonderEditor from "../../../../../../tool/engine/NoWorkerJobConfigToolEngine.js";

describe("MainEditorTransform scale", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                return /* () */0;
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        MainEditorTransformTestTool$WonderEditor.transformBaseTest(sandbox, "test change scale value", /* tuple */[
              1,
              TransformUtils$WonderEditor.getTransformScaleData
            ], /* tuple */[
              MainEditorTransformTool$WonderEditor.changeScaleX,
              MainEditorTransformTool$WonderEditor.changeScaleY,
              MainEditorTransformTool$WonderEditor.changeScaleZ
            ]);
        describe("deal with specific case", (function () {
                beforeEach((function () {
                        MainEditorSceneTool$WonderEditor.initStateWithJob(sandbox, NoWorkerJobConfigToolEngine$WonderEditor.buildNoWorkerJobConfig(undefined, undefined, undefined, undefined, undefined, /* () */0), undefined, undefined, undefined, /* () */0);
                        MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                        return DirectorToolEngine$WonderEditor.prepareAndInitAllEnginState(/* () */0);
                      }));
                describe("the scale value in engineState can't be 0", (function () {
                        return Wonder_jest.test("if input 0, set origin value to engineState instead of 0", (function () {
                                      GameObjectTool$WonderEditor.getCurrentSceneTreeNodeTransform(/* () */0);
                                      var reasonStateUpdate = ReactTool$WonderEditor.getUpdateState(FloatInputTool$WonderEditor.reducer(undefined, undefined, true, /* Blur */0, /* record */[
                                                /* inputValue */"0.",
                                                /* originValue */"1.0"
                                              ], /* () */0));
                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](reasonStateUpdate[/* inputValue */0]), "0");
                                    }));
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
