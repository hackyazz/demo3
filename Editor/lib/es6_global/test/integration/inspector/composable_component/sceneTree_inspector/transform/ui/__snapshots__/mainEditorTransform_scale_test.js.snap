// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorTransform scale test change scale value changeX should set current gameObject->transform->value x set x value to floatInput 1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="-10.1213"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;
